### PR TITLE
Analytics: Clean up analytics, warn when sending nested props

### PR DESCRIPTION
**Pull request by blowery**
Nested props are not supported by Tracks. Warn when not in production that we're trying to send them.

Also clean up analytics. Commits split to make the review easier.

Test live: https://calypso.live/?branch=update/tracks/detect-nested-props

### CODE REVIEW

File: client/lib/analytics/index.js [hunk 1]
```diff
@@ -61,15 +63,16 @@ function buildQuerystringNoPrefix( group, name ) {
...
-var mostRecentUrlPath = null;
-
-window.addEventListener('popstate', function() {
-	// throw away our URL value if the user used the back/forward buttons
-	mostRecentUrlPath = null;
-});
+let mostRecentUrlPath = null;
 
-var analytics = {
+if ( typeof window !== 'undefined' ) {
```
Comment (TooTallNate) [inline client/lib/analytics/index.js]: Is the check here because of server-side rendering?

Comment (blowery) [inline client/lib/analytics/index.js]: yup and for test mocking. it's a pain to run the tests in node and have to fake out window just to get the tests to load.

**Comment by lancewillett** [comment]
This looks pretty sane — @blowery can we merge or does it need more review / testing?

**Comment by gwwar** [comment]
@blowery if you rebase I can 👀

**Comment by blowery** [comment]
@gwwar rebased

File: client/lib/analytics/index.js [hunk 2]
```diff
@@ -122,18 +125,27 @@ var analytics = {
...
 	tracks: {
 		recordEvent: function( eventName, eventProperties ) {
-			var superProperties;
+			let superProperties;
 
 			eventProperties = eventProperties || {};
 
+			if ( process.env.NODE_ENV !== 'production' ) {
+				for ( const key in eventProperties ) {
+					if ( isObjectLike( eventProperties[ key ] ) && typeof console !== 'undefined' ) {
```
Comment (gwwar) [inline client/lib/analytics/index.js]: We need to disable the eslint rule here:
```
// eslint-disable-line no-console
```

**Pull request merged**